### DISCUSSION TITLE
Widget for Android

### DISCUSSION THREAD

**Discussion by edge7** [Radiokot/photoprism-android-client#98] [Ideas]
👉🏻 [Start discussing an idea](https://github.com/Radiokot/photoprism-android-client/discussions/new?category=ideas)
Hi,

is there any plan to add a widget, similar to what Google Photo does? Something like: it happened one year ago, with the picture?

Cheers

**Comment by Radiokot** [comment]
Good evening, Enrico.
I am planning to add "Happened ago" feature as a notification, but never thought about a widget. Although I've never used Google Photos widget, the idea seems good to me. 
Thank you.

**Reply by edge7** [reply]
Great. Thanks you.
when the 'happened x ago', should be expected, roughly?

**Reply by Radiokot** [reply]
I don't know, maybe this year 🤷🏻‍♂️

**Comment by edge7** [comment]
Hey.

Any update on this? It would be very nice to have it!

**Reply by Radiokot** [reply]
Hello again, Enrico. 
Not yet.

**Reply by edge7** [reply]
Roughly by end of this year? :)

**Reply by Radiokot** [reply]
<img src="https://github.com/Radiokot/photoprism-android-client/assets/5675681/8700e2e6-3792-4a62-8364-9cf3010e7cee" width=300 />

**Comment by Radiokot** [comment]
Enrico, here's a clone gallery build in which you can test the Memories feature. If there is something in your library for this day in the past, the memory card is shown on the main screen. The memories are updated daily around 08:30, currently there are no notifications for them. To make the instant update, hold the search bar to get onto the dev screen and click "Update now".
https://send.bitwarden.com/#fzXZxNAsp0SmIbEjAU-wQQ/nhfMlkhEBKDrsVPn515fiA
<img width=300 src="https://github.com/Radiokot/photoprism-android-client/assets/5675681/ad282392-b6a2-4e12-9017-655a54aac8f5" />

**Comment by edge7** [comment]
Hey. Just seen this message.

Thanks for the effort. Am testing it out and will give feedback.

Thanks again

**Comment by edge7** [comment]
Hey,
thanks again for the feature.
It works fine for me, it has shown the right pictures, on the right days.
Also, no crash or anything like that. 

I am far from being an app/ui designer, but I believe the 'update now' button is not required. Even without that, the experience would be fine.

Am looking forward to test a widget or notification system!! :)

**Reply by Radiokot** [reply]
It's a development-only screen, no one will have to press buttons once it is live 😅  Thanks for testing.

**Reply by Radiokot** [reply]
Hi there, here is a clone version with notifications for new memories. Can be enabled/disabled from Preferences -> Extensions
https://send.bitwarden.com/#r0AVaGeDHUyyQLEyAUGaBQ/J25mQHvphbhvBc3zz77YjQ

**Comment by Radiokot** [comment]
Hi. Have you seen the recent updates? The feature (without widget) is available as a gallery extension called Memories. You can learn more about it here: https://github.com/Radiokot/photoprism-android-client/releases/tag/1.27.0(41)

**Reply by edge7** [reply]
How can I get a key to enable extensions?
Can I please pay in XMR (Monero) and not BTC?

**Reply by Radiokot** [reply]
Of course, please use the Help option on the extension activation screen (Preferences > Extensions) to write me an email. I need the included identifier to issue a key.
My Monero address is `8B2kHzxp36nVdk2UiZqkaW14LuRrt2wZPRmFXEr2EvRyA2marKXeDfcB1Mv6w5ZM9pFtPuqqqFc61b7Muc9rkXJjQmhMBfF`
